### PR TITLE
Implement ModifyInstanceAttribute

### PR DESCRIPTION
**Pull request by axw**
This PR introduces the ModifyInstanceAttribute action, for a subset of possible attributes. We currently support altering SourceDestCheck and BlockDeviceMappings.

### CODE REVIEW

File: ec2/ec2.go [hunk 1]
```diff
@@ -560,6 +560,47 @@ func (ec2 *EC2) Instances(instIds []string, filter *Filter) (resp *InstancesResp
...
+// ModifyInstanceAttribute modifies a single attribute of an instance in EC2.
+//
+// See http://goo.gl/cTxLa7 for more details.
+func (ec2 *EC2) ModifyInstanceAttribute(req *ModifyInstanceAttribute, filter *Filter) (resp *SimpleResp, err error) {
+	params := makeParams("ModifyInstanceAttribute")
+	params["InstanceId"] = req.InstanceId
+	if req.SourceDestCheck != nil {
+		params["SourceDestCheck.Value"] = fmt.Sprint(*req.SourceDestCheck)
+	}
+	if req.BlockDeviceMappings != nil {
```
Comment (dimitern) [inline ec2/ec2.go:582]: The way this is written implies you could set both SourceDestCheck and BlockDeviceMappings, but according to the AWS API docs you can only set one at a time.

File: ec2/volumes_test.go [hunk 2]
```diff
@@ -304,3 +304,12 @@ func assertVolumeAttachment(c *C, obtained *ec2.VolumeAttachmentResp, volId, ins
...
 	c.Check(obtained.Status, Matches, "(attaching|attached)")
 }
+
+func deleteVolume(c *C, e *ec2.EC2, id string) {
+	_, err := e.DeleteVolume(id)
+	if err != nil && c.Check(err, ErrorMatches, ".*InvalidVolume.NotFound.*") {
+		// Nothing to do.
+		return
+	}
+	c.Assert(err, IsNil, Commentf("%v LEFT RUNNING!!!", id))
```
Comment (dimitern) [inline ec2/volumes_test.go]: I know this is probably a copy/paste thing, but the comment could be better when we're talking about deleting volumes.

**Comment by dimitern** [comment]
Great work! LGTM with a couple of suggestions. Thanks for doing the SourceDestCheck flag and tests!

File: ec2/volumes_test.go (hunk 2 above)
Comment (axw) [inline ec2/volumes_test.go]: Reworded.

File: ec2/ec2.go (hunk 1 above)
Comment (axw) [inline ec2/ec2.go:582]: Yeah, I'm relying on the server to return an error. I don't think it's really worth adding code here to check as well.

**Pull request merged**